runtime/vm/createramclass.cpp

```diff
@@ -2081,6 +2083,20 @@ checkFlattenableFieldValueClasses(J9VMThread *currentThread, J9ClassLoader *clas
 					result = FALSE;
 					break;
 				}
+
+				/* A NullRestricted field must be in a value class with an
+				 * ImplicitCreation attribute. The attribute must have the ACC_DEFAULT flag set.
+				 * Static fields will be checked during class preparation.
+				 */
+				if (J9_ARE_ALL_BITS_SET(modifiers, J9FieldFlagIsNullRestricted)) {
```

Comment:
Forgot this important line here :) the functional/Valhalla tests are passing for me locally now.